### ISSUE TITLE
Watch on uiSortable should handle deleted values

### ISSUE THREAD

**Issue by tepez** [angular-ui/ui-sortable#345]
Currently, the [watch](https://github.com/angular-ui/ui-sortable/blob/v0.14.x-dev/src/sortable.js#L297) on uiSortable only checks for properties of `newVal`. 

As a result, we never update the sortable instance when a property is deleted, i.e. it was in `prevVal` but notin `newVal`.

**Comment by thgreasi** [comment]
Oh I see... This should be easy.
Does setting null as the new value seems correct thought?

**Comment by thgreasi** [comment]
Do you have a use case that is currently broken because of this? Off this is not high priority , then we can mark it for v0.14.x.

**Comment by tepez** [comment]
Don't have any use case for it,  just noticed it when I worked on the extra elements PR. I would say this is rather low priority.

I think the expected behavior when an opiton (appendTo, delay...)  appears in `oldVal` but not in `newVal` is reset to the default of jquery-ui-sortable, as if the directive was initialized without that option. We can get the default values by creating a dummy sortable instance for all the sortable directives, e.g.:
```js
var sortableInstance = jQuery.ui.sortable();
// default options are now at `sortableInstance.options`
```
`null` is a valid value for most options, different from the default value so I don't think it's expected.

For a callback (start, stop...) I would just re-combining it with null as the callback.

BTW, the code for combining a callback is different  at [initialization](https://github.com/angular-ui/ui-sortable/blob/v0.14.x-dev/src/sortable.js#L335) and in the [watch](https://github.com/angular-ui/ui-sortable/blob/v0.14.x-dev/src/sortable.js#L315). Is this intentional?
Also, `helper` is only wrapped in the [watch](https://github.com/angular-ui/ui-sortable/blob/v0.14.x-dev/src/sortable.js#L325) and not on initialization.

**Comment by thgreasi** [comment]
The later options initialization loop was introduced to bypass a jquery-ui problem when changing the value of some options at a later time.
Since the $watch loop is executed right after `element.sortable(opts);` takes place, I chose to set only the required subset of options, so that less work is done.

We could probably refactor that code to make it look better :)

**Comment by thgreasi** [comment]
Fixed in v0.14 by #373
